src/site/content/en/blog/wakelock/index.md

```diff
@@ -0,0 +1,264 @@
+  just the way `AbortController` works.
+* The browser can also refuse the request for different reasons, for example,
+  because the battery charge level is too low. In this case, the exception's
+  message will contain more details.
+
+```js
+const wakeLockCheckbox = document.querySelector('#wakeLockCheckbox');
+
+if ('WakeLock' in window) {
+  let wakeLock = null;
+
+  const requestWakeLock = () => {
+    const controller = new AbortController();
+    const signal = controller.signal;
+    window.WakeLock.request('screen', {signal})
+        .catch((e) => {
+          if (e.name === 'AbortError') {
+            wakeLockCheckbox.checked = false;
+            console.log('Wake Lock was aborted');
+          } else {
+            console.error(`${e.name}, ${e.message}`);
+          }
+        });
+    wakeLockCheckbox.checked = true;
+    console.log('Wake Lock is active');
+    return controller;
+  };
+
+  wakeLockCheckbox.addEventListener('change', () => {
+    if (wakeLockCheckbox.checked) {
+      wakeLock = requestWakeLock();
+    } else {
+      wakeLock.abort();
+      wakeLock = null;
+    }
+  });
+}
+```
+
+### The wake lock lifecycle {: #wake-lock-lifecycle }
+
+When you play with the [wake lock demo][demo], you'll notice that wake locks
+are sensitive to [page visibility][page-visibility-api], as well as
+[full screen changes][full-screen-api]. This means that when you minimize a
+tab or window where a wake lock is active, or enter full screen mode, the
+wake lock will automatically abort. To re-acquire the wake lock,
+listen for the events that either of the APIs emit,
+namely the [`visibilitychange`][visibility-change] event of the
+Page Visibility API, and the [`fullscreenchange`][fullscreen-change] event
+of the Fullscreen API, and then request a new wake lock.
+
+```js
+const handleVisibilityChange = () => {
+  if (wakeLock !== null && document.visibilityState === 'visible') {
+    wakeLock = requestWakeLock();
+  }
+};
+
+document.addEventListener('visibilitychange', handleVisibilityChange);
+document.addEventListener('fullscreenchange', handleVisibilityChange);
+```
+
+## Best Practices {: #best-practices }
+
+The approach you take depends on the needs of your app. Regardless, you should
+use the most lightweight approach possible for your app to minimize your app's
+impact on system resources.
+
+Before adding wake lock to your app, consider whether your use cases could
+be solved with one of the following alternative solutions:
+
+* If your app is performing long-running downloads, consider using
+  [background fetch](https://developers.google.com/web/updates/2018/12/background-fetch).
+* If your app is synchronizing data from an external server, consider using
+  [background sync](https://developers.google.com/web/updates/2015/12/background-sync).
+
+{% Aside %}
+  Like most other powerful web APIs, the Wake Lock API is only available
+  when served over **HTTPS**.
```

Comment:
We've done that pretty consistently to call out HTTPS being required. I like making it clear that its **HTTPS**

Reply:
This kinda makes my editor sense tingle because we tend to use bolding semantically (e.g., to indicate the name of a UI control), and I think having a one-sentence aside makes "HTTPS" pretty salient. But there are other, higher hills to die on!😄